File: semester_5/ON/L2/ex2.jl
#Paweł Grzegory 282211

using Plots

f(x) = exp(x) * log(1 + exp(-x))

# Zakres x
x = -100:0.1:100

y = f.(x)

plot(x, y,
    label="f(x) = exp(x) * log(1 + exp(-x))",
    xlabel="x",
    ylabel="f(x)",
    title="Wykres funkcji f(x)",
    legend=:topleft,
    grid=true
)

savefig("wykres_julia.png")
println("Wykres został zapisany do pliku wykre_julia.png")

File: semester_5/ON/L2/ON-Lab2-Paweł-Grzegory/ex2.py
#Paweł Grzegory 282211

import numpy as np
import matplotlib.pyplot as plt

def f(x):
    return np.exp(x) * np.log(1 + np.exp(-x))

x = np.arange(-100, 100, 0.1)

y = f(x)

plt.figure(figsize=(10, 6))
plt.plot(x, y, label="f(x) = exp(x) * log(1 + exp(-x))")
plt.xlabel("x")
plt.ylabel("f(x)")
plt.title("Wykres funkcji f(x)")
plt.legend()
plt.grid(True)

plt.savefig("wykres_python.png")
print("Wykres został zapisany do pliku wykres_python.png")
